src/App.tsx

```diff
@@ -1,44 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPostsData, getUsersData } from './api/api';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [loader, setLoader] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
```

Comment:
please add the correct naming because now it's quite confusing to understand what the purpose of this state
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false);
```